custom/icds/messaging/indicators.py

```diff
@@ -238,8 +240,13 @@ class AWWSubmissionPerformanceIndicator(AWWIndicator):
     def __init__(self, domain, user):
         super(AWWSubmissionPerformanceIndicator, self).__init__(domain, user)
 
-        result = get_last_submission_time_for_users(self.domain, [self.user.get_id], self.get_datespan())
-        self.last_submission_date = result.get(self.user.get_id)
+        span = self.get_datespan()
+        self.last_submission_date = ApplicationStatusFact.objects.filter(
+            user_dim__user_id=user.get_id,
+            domain=domain,
+            last_form_submission_date__gte=span.startdate.date(),
+            last_form_submission_date__lte=span.enddate.date(),
+        ).aggregate(value=Max("last_form_submission_date"))["value"]
```

Comment:
If you can change this to query for awc_id instead of user id, you can use https://github.com/dimagi/commcare-hq/blob/fe885ffedb943c7b559db2429a902299238d8b4a/custom/icds_reports/models/aggregate.py#L1183

As of now there should be a one-to-one mapping of awc_id and user

Reply:
Err. actually that may be slightly different behavior. This would give you the last submission for the following XMLNS:

https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/ucr/data_sources/usage_forms.json#L28-L41

That table is consistent with what Neal has been using to generate an "inactive AWW" report, so it could be used here, but I believe that it doesn't include all forms in the app.

Reply:
hmm actually i'm not sure we should filter on specific forms. the message being sent in response to this inactivity says "... has not submitted any form or done any activity in ICDS CAS application ..." so it might be confusing to someone receiving the message if we don't check all form submissions.

Reply:
I agree with @gcapalbo, I'd prefer not to filter on specific forms for future-proof-ness and because of the message content.

@emord said
> This would give you the last submission for the following XMLNS: ...

I'm not sure what you meant by "this" in that sentence? Are you saying the code I've written will only check a subset of forms in the app (the ones with those XMLNS you linked to)? If that's true, then I think I want to change this, but I think that's not what you meant.

I'm planning to leave this as is unless you can clarify why it's a bad idea to do that. FWIW, I didn't catch it if you already tried to communicate that.

Reply:
Sorry that was unclear. "This" was referring to the change I was suggesting. I think this is a fine change as is (once app status fact is ready to be used).

FWIW we currently have two definitions of "inactive" AWW in code right now. This SMS indicator which considers all forms and another which only considers the 12 XMLNS that is the list I linked. I'll spin up an email thread on this with you guys